core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -68,4 +57,58 @@ public void end() {
         LOG.debug("Shutting down consumer for topic {}", topic);
         rawAdapter.unsubscribe();
     }
+
+    public <A1> Consumer on(Event event, SingleArgEventHandler<A1> eventHandler) {
+        eventEmitter.on(event, eventHandler);
+        return this;
+    }
+
+    public <A1, A2> Consumer on(Event event, TwoArgsEventHandler<A1, A2> eventHandler) {
+        eventEmitter.on(event, eventHandler);
+        return this;
+    }
+
+    public <A1> Consumer emit(Event event, A1 arg) {
+        eventEmitter.emit(event, arg);
+        return this;
+    }
+
+    protected void handleRawMessage(String jsonMessage) {
+        LOG.debug("Topic [{}] message received [{}]", this.topic, jsonMessage);
+        Exception error = null;
+        Message message = null;
+
+        try {
+            if (msbConfig.getSchema() != null
+                    && !Utils.isServiceTopic(topic)) {
+                Utils.validateJsonWithSchema(jsonMessage,
+                        msbConfig.getSchema());
+            }
+            message = Utils.fromJson(jsonMessage,
+                    Message.class);
+        } catch (JsonConversionException | JsonSchemaValidationException e) {
+            error = e;
+        }
+
+        if (error != null) {
+            eventEmitter.emit(ERROR_EVENT, error);
+            return;
+        }
+
+        if (isMessageExpired(message))
+            return;
+
+        if (channelMonitorAgent != null) {
```

Comment:
This is a redundant check. Channel manager is never null in Channel manager so you can just put validation statement in constructor